src/passes/hash-stringify-walker.cpp

```diff
@@ -55,8 +55,8 @@ bool StringifyEquator::operator()(Expression* lhs, Expression* rhs) const {
 void HashStringifyWalker::addUniqueSymbol() {
   // Use a negative value to distinguish symbols for separators from symbols
   // for Expressions
-  hashString.push_back((uint64_t)-nextVal);
-  nextVal++;
+  hashString.push_back((unsigned)nextSeparatorVal);
+  nextSeparatorVal--;
```

Comment:
We could be extra safe here by asserting that the negative numbers and positive numbers have not crossed each other and started accidentally repeating values.

Reply:
done